src/Traversals/dfs.jl

```diff
@@ -2,52 +2,67 @@ import Base.showerror
 struct CycleError <: Exception end
 Base.showerror(io::IO, e::CycleError) = print(io, "Cycles are not allowed in this function.")
 
-struct DFS <: TraversalAlgorithm end
+struct DepthFirst <: TraversalAlgorithm
+    neighborfn::Function
```

Comment:
This should be parameterized by the function type
```julia
struct DepthFirst{F}
    neighborfn::F
```

Reply:
Absolutely right, now that I think about it, since `Function` is abstract.

Reply:
also, not sure you should impose the subtyping on function, since many things can be callable without being subtypes of functions